test/test_type_promotion.py

```diff
@@ -329,8 +329,7 @@ def test_alternate_result(self, device):
         d = torch.tensor([1, 1, 1, 1], dtype=torch.double, device=device)
         torch.add(f, f, out=d)
         self.assertEqual(d.dtype, torch.double)
-        # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-        self.assertEqualIgnoreType(f + f, d)
+        self.assertEqual(f + f, d, exact_dtype=False)
```

Comment:
We don't want to just add `exact_dtype=False` - this should be the last resort only in there is no way to align the types in other ways.